tests/cloudpickle_test.py

```diff
@@ -2321,6 +2326,32 @@ def __type__(self):
         o = MyClass()
         pickle_depickle(o, protocol=self.protocol)
 
+    @pytest.mark.skipif(
+        sys.version_info < (3, 6, 0),
+        reason="Dict determinism is a lost cause in Python < 3.6")
+    def test_sorted_globals(self):
+        vals = set()
+
+        def func_with_globals():
+            return _TEST_GLOBAL_VARIABLE + _TEST_GLOBAL_VARIABLE2
+
+        for i in range(5):
+            vals.add(
+                subprocess_pickle_string(func_with_globals,
+                                         protocol=self.protocol,
+                                         add_env={"PYTHONHASHSEED": str(i)}))
+        assert len(vals) == 1
+
+    def test_efficient_sorted_globals(self):
+        # Non regression test to demonstrate that large numbers of globals
+        # do not cause slowdown
+        gvars = set(f"a{i}" for i in range(1000))
+        assert cloudpickle.cloudpickle._extract_code_globals(
+            _TEST_BIG_GLOBAL_SPACE.__code__) == gvars
+        tic = time.time()
+        subprocess_pickle_string(_TEST_BIG_GLOBAL_SPACE, protocol=self.protocol)
+        assert time.time() - tic < 0.5
```

Comment:
I would rather not put tests that depend on the runtime because the CI workers can overloaded from time to time and those tests would fail at random.

Instead just run a one-off benchmark and post the results in the comments of this PR.

Reply:
No need to address this if we chose to go for #428 (which I think we should ;).